### DISCUSSION TITLE
Editing the .jl file outside of Pluto

### DISCUSSION THREAD

**Discussion by sje30** [fonsp/Pluto.jl#1314] [Questions about using Pluto] [answered]
Hi,

I like the notion that the source file for Pluto is just a plain .jl file that I can open in my regular editor (Emacs).  Is it possible to edit the file in the editor and then reload in Pluto, or are the ids for each chunk going to be invalid?  Likewise, can I add new code chunks, or should I assume that this will break all the chunk names?  Is there anyway that they can be reset in Pluto?

**Comment by lungben** [comment]
It should be fine if you only modify existing cells but not add new cells / delete cells (or add / remove packages if using PlutoPkg).
The cell names are just UUIDs, they do not contained hashed cell content or similar.
The cell execution order should be recalculated when you open the modified file again in Pluto, you may get a warning message if it has changed but it should be fine (however, I have not tested it yet).

But in general this workflow is not recommended.

**Reply by sje30** [reply]
Thank you, good to know.

**Comment by Pangoraw** [comment, marked as answer]
Hi! You may also be interested by [this pull-request](https://github.com/fonsp/Pluto.jl/pull/1029) which enables editing the underlying notebook file and directly push updates to Pluto.

**Reply by sje30** [reply]
that is cool.
